python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -664,247 +667,288 @@ def run(
-                    columns_target[f'mag_{column_target_flux}'] = (
-                        column_mag_temp, column_mag_temp_model, None, False,
-                    )
-
-                    # No need for colors if this is the last band and there are only two bands
-                    # (because it would just be the negative of the first color)
-                    skip_color = (idx_band == n_bands) and (n_bands <= 2)
-                    if not skip_color:
-                        column_color_temp_model = f'{column_color_temp}{idx_model}'
-                        column_color_err_temp_model = f'{column_color_err_temp}{idx_model}'
-
-                        # e.g. if order is ugrizy, first color will be u - g
-                        cat_target[column_color_temp_model] = mag_prev[idx_model] - mag_model[idx_model]
-
-                        # Sum (in quadrature, and admittedly sketchy for faint fluxes) magnitude errors
-                        cat_target[column_color_err_temp_model] = 2.5/np.log(10)*np.hypot(
-                            flux_err_frac[idx_model], flux_err_frac_prev[idx_model])
-                        columns_target[f'color_{band_prev}_m_{band}_{column_target_flux}'] = (
-                            column_color_temp,
-                            column_color_temp_model,
-                            column_color_err_temp_model,
-                            False,
-                        )
-
-                    for idx_bin, (mag_lo, mag_hi) in enumerate(bins_mag):
-                        row = data[idx_bin]
-                        # Reference sources only need to be counted once
-                        if idx_model == 0:
-                            select_ref_bin = select_ref_bins[idx_bin]
-                        select_target_bin = select_target & (mag_target > mag_lo) & (mag_target < mag_hi)
-
-                        for sourcetype in SourceType:
-                            sourcetype_info = sourcetype.value
-                            is_extended = sourcetype_info.is_extended
-                            # Counts filtered by match selection and magnitude bin
-                            select_ref_sub = select_ref_bin.copy()
-                            select_target_sub = select_target_bin.copy()
-                            if is_extended is not None:
-                                is_extended_ref = (extended_ref == is_extended)
-                                select_ref_sub &= is_extended_ref
+        if config.include_unmatched:
+            # Create an unmatched table with the same schema as the matched one
+            # ... but only for objects with no matches (for completeness/purity)
+            # and that were selected for matching (or inclusion via config)
+            cat_right = apTab.Table.from_pandas(cat_ref[~matched_ref & select_ref].reset_index(drop=False))
+            cat_right.rename_columns(cat_right.colnames, [f"refcat_{col}" for col in cat_right.colnames])
```

Comment:
There's a mix of `'` and `"` in this script. I wouldn't be averse to you running the whole thing through `black` prior to a merge to resolve this (and other formatting choices).

Reply:
Hmm... I'll consider it depending on how many changes there are. Sometimes I wish older pipelines packages would adopt middleware/atools-like pre-commit and do a run through.